### PR TITLE
Alibi Update

### LINKED ISSUE #1813: SKLearn version support too low
The following is my sklearn_server.yaml
```
apiVersion: machinelearning.seldon.io/v1alpha2
kind: SeldonDeployment
metadata:
  name: sklearn
  namespace: seldon-system
spec:
  #annotations:
  #  seldon.io/executor: "true"
  name: iris-s3
  predictors:
    - graph:
        children: []
        implementation: SKLEARN_SERVER
        modelUri: s3://mymodel
        envSecretRefName: seldon-init-container-secret
        name: classifier
      name: default
      replicas: 1
```

I upload a sklearn model into S3 storage. The Seldon classifier fails with ModuleNotFoundError: No module named 'sklearn.svm._classes'

```
Traceback (most recent call last):
  File "/opt/conda/bin/seldon-core-microservice", line 11, in <module>
    load_entry_point('seldon-core', 'console_scripts', 'seldon-core-microservice')()
  File "/microservice/python/seldon_core/microservice.py", line 288, in main
    user_object = user_class(**parameters)
  File "/microservice/SKLearnServer.py", line 23, in __init__
    self.load()
  File "/microservice/SKLearnServer.py", line 31, in load
    self._joblib = joblib.load(model_file)
  File "/opt/conda/lib/python3.7/site-packages/joblib/numpy_pickle.py", line 605, in load
    obj = _unpickle(fobj, filename, mmap_mode)
  File "/opt/conda/lib/python3.7/site-packages/joblib/numpy_pickle.py", line 529, in _unpickle
    obj = unpickler.load()
  File "/opt/conda/lib/python3.7/pickle.py", line 1088, in load
    dispatch[key[0]](self)
  File "/opt/conda/lib/python3.7/pickle.py", line 1376, in load_global
    klass = self.find_class(module, name)
  File "/opt/conda/lib/python3.7/pickle.py", line 1426, in find_class
    __import__(module, level=0)
ModuleNotFoundError: No module named 'sklearn.svm._classes'
```

I ask in the slack community, thanks Clive Cox tell me that seldon seems support sklearn v0.20.3 for now.

### LINKED ISSUE #2144: SeldonDeployment explainer description
Hi,
I am new to seldon, read through the documentation (may have overlooked it) and I have not found a description for the **explainer** part of the SeldonDeployment file. So i got some questions regarding it, but first if there is a documentation could someone point me to it?

For example in this file:
```
apiVersion: machinelearning.seldon.io/v1
kind: SeldonDeployment
metadata:
  name: income
spec:
  name: income
  annotations:
    seldon.io/rest-timeout: "100000"
  predictors:
  - graph:
      children: []
      implementation: SKLEARN_SERVER
      modelUri: gs://seldon-models/sklearn/income/model
      name: classifier
    explainer:
      type: AnchorTabular
      modelUri: gs://seldon-models/sklearn/income/explainer
    name: default
    replicas: 1
```

Regarding the explainer:  
1) Is the **type** just alibi specific or can also a custom type/explainer_model be used?
2) To what points the **modelUri**? I read that *AnchorTabular* is a local explanation is, therefor no separate explanation_model can be trained. Does that mean the **modelUri** points to a (for example) python file with the interpretability logic?
3) Is there always a **modelUri** needed? I saw an example where none was specified, why?

### LINKED ISSUE #2160: Upgrade Alibi Integrations
Alibi Wrapper:

 * Alibi to latest release 0.5
 *  Ensure dill is upgraded to 0.3.2 or look at latest Alibi explort/load capabilities

Alibi Detect Wrapper:
 * Alibi-detect to latest release 0.4.1
 * Handle classes that don't allow `outlier_type` in `predict`
 * Expose events as metrics that can be scrapped
 * Look at request logger integration

### PR DESCRIPTION
**Pull request by cliveseldon**
**What this PR does / why we need it**:
Update Alibi integrations

**Which issue(s) this PR fixes**:

Fixes #2160
Fixes #2144
Fixes #1813 

 * Brings all explainer source code to inside seldon so kfserving is not a dependency
 * Adds Integrated gradients and KernelShap from Alibi 0.5.2
 * Adds initial ALE from Alibi 0.5.2
 * Adds docs on explainers
 * Extends explainers notebook to include kernel_shap and integrated gradients
 * Updates scikit-learn to 0.23.2
 * Adds docs on using custom or older version of scikit-learn

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Adds further documentation for using explainers
Update scikit-learn version to 0.23.2 and updates docs to discuss how you can continue to use older or custom versions.
```

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **cliveseldon**
You can assign the PR to them by writing `/assign @cliveseldon` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
Tue Aug  4 09:00:17 UTC 2020
The logs for [pr-build] [1] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2241/1.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2241 --build=1

**Comment by seldondev** [comment]
Tue Aug  4 09:00:29 UTC 2020
The logs for [lint] [2] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2241/2.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2241 --build=2

**Comment by seldondev** [comment]
Tue Aug 11 08:43:34 UTC 2020
The logs for [pr-build] [3] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2241/3.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2241 --build=3

**Comment by seldondev** [comment]
Tue Aug 11 08:43:58 UTC 2020
The logs for [lint] [4] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2241/4.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2241 --build=4

**Comment by seldondev** [comment]
@cliveseldon: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | 79fd324c9fc9234382832d860a5d1d1c072cdd55 | [link]() | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
Wed Aug 12 11:54:10 UTC 2020
The logs for [pr-build] [5] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2241/5.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2241 --build=5

**Comment by seldondev** [comment]
@cliveseldon: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | e14509bd56f4773c172c436f892e59ad2f7c65a5 | [link]() | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
Sat Aug 15 08:39:52 UTC 2020
The logs for [pr-build] [7] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2241/7.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2241 --build=7

**Comment by seldondev** [comment]
Sat Aug 15 08:40:03 UTC 2020
The logs for [lint] [8] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2241/8.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2241 --build=8

**Comment by seldondev** [comment]
Mon Aug 17 14:34:22 UTC 2020
The logs for [lint] [10] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2241/10.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2241 --build=10

**Comment by seldondev** [comment]
Mon Aug 17 14:34:23 UTC 2020
The logs for [pr-build] [9] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2241/9.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2241 --build=9

**Comment by seldondev** [comment]
Mon Aug 24 07:36:33 UTC 2020
The logs for [lint] [16] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2241/16.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2241 --build=16

**Comment by seldondev** [comment]
Mon Aug 24 07:36:53 UTC 2020
The logs for [pr-build] [15] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2241/15.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2241 --build=15

**Comment by seldondev** [comment]
Mon Aug 24 07:38:00 UTC 2020
The logs for [integration] [17] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2241/17.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2241 --build=17

**Comment by seldondev** [comment]
@cliveseldon: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint | 24e76a53ec6000707f1a9841d51faf6cea5d22b8 | [link]() | `/test lint`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
Mon Aug 24 08:33:01 UTC 2020
The logs for [pr-build] [18] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2241/18.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2241 --build=18

**Comment by seldondev** [comment]
Mon Aug 24 08:33:00 UTC 2020
The logs for [lint] [19] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2241/19.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2241 --build=19

**Comment by seldondev** [comment]
@cliveseldon: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint | 23b573b686907ab8e5fef1b14eb419d0773cedfc | [link]() | `/test lint`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint | 23b573b686907ab8e5fef1b14eb419d0773cedfc | [link]() | `/test lint`
integration | 24e76a53ec6000707f1a9841d51faf6cea5d22b8 | [link]() | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
Mon Aug 24 12:51:58 UTC 2020
The logs for [pr-build] [20] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2241/20.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2241 --build=20

**Comment by seldondev** [comment]
Mon Aug 24 12:52:13 UTC 2020
The logs for [lint] [21] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2241/21.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2241 --build=21

**Comment by seldondev** [comment]
Mon Aug 24 13:14:45 UTC 2020
The logs for [integration] [22] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2241/22.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2241 --build=22

**Comment by seldondev** [comment]
Mon Aug 24 13:16:03 UTC 2020
The logs for [notebooks] [23] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2241/23.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2241 --build=23

**Comment by seldondev** [comment]
@cliveseldon: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 24e76a53ec6000707f1a9841d51faf6cea5d22b8 | [link]() | `/test integration`
pr-build | 9e80fa2878854f2fe9ac13a3f26a88618f0ec9e9 | [link]() | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | 9e80fa2878854f2fe9ac13a3f26a88618f0ec9e9 | [link]() | `/test this`
integration | 9e80fa2878854f2fe9ac13a3f26a88618f0ec9e9 | [link]() | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | 9e80fa2878854f2fe9ac13a3f26a88618f0ec9e9 | [link]() | `/test this`
integration | 9e80fa2878854f2fe9ac13a3f26a88618f0ec9e9 | [link]() | `/test integration`
notebooks | 9e80fa2878854f2fe9ac13a3f26a88618f0ec9e9 | [link]() | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
Mon Aug 24 16:17:45 UTC 2020
The logs for [pr-build] [24] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2241/24.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2241 --build=24

File: components/alibi-explain-server/alibiexplainer/explainer.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+        super().__init__(name)
+        self.predictor_host = predictor_host
+        logging.info("Predict URL set to %s", self.predictor_host)
+        self.method = method
+        self.protocol = protocol
+        self.tf_data_type = tf_data_type
+        logging.info("Protocol is %s",str(self.protocol))
+
+        if self.method is ExplainerMethod.anchor_tabular:
+            self.wrapper:ExplainerWrapper = AnchorTabular(self._predict_fn, explainer, **config)
```
Comment (axsaucedo) [inline components/alibi-explain-server/alibiexplainer/explainer.py:86]: Adds extra ExplainerWrapper type only on this one

File: components/alibi-explain-server/alibiexplainer/explainer.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+        pass
+
+    def _predict_fn(self, arr: Union[np.ndarray, List]) -> np.ndarray:
+        print(arr)
+        if type(arr) == list:
+            arr = np.array(arr)
+        if self.protocol == Protocol.seldon_grpc:
+            return self._grpc(arr)
+        elif self.protocol == Protocol.seldon_http:
+            payload = seldon.create_request(arr, seldon.SeldonPayload.NDARRAY)
```
Comment (axsaucedo) [inline components/alibi-explain-server/alibiexplainer/explainer.py:108]: Why not use the SeldonClient, is the current request data structre not supported?

**Review by axsaucedo — COMMENTED** [review]
Added a couple of comments - only other thing is that it would be quite important to have the code used to generate some of the trained explainers in the example also as notebooks (not sure if they are in the kfserving repo or here - if the former we could link them in the docs)?

File: doc/source/analytics/explainers.md [hunk 3]
```diff
@@ -7,4 +7,50 @@ Seldon provides model explanations using its [Alibi](https://github.com/SeldonIO
 
 We provide [an example notebook](../examples/explainer_examples.html) showing how to deploy an explainer for Tabular, Text and Image models.
 
+## Creating your explainer
 
+For Alibi explainers that need to be trained you should
+
+ 1. Use python 3.6 as the Seldon python wrapper also runs in python 3.6 when it loads your explainer.
```
[the remark this answers is not in the document]
Comment (cliveseldon) [reply doc/source/analytics/explainers.md:14]: yeah - there is no ubi 3.7 image at present. There is 3.8 but dill has issues with that.

File: components/alibi-explain-server/alibiexplainer/explainer.py (hunk 1 above)
Comment (cliveseldon) [reply components/alibi-explain-server/alibiexplainer/explainer.py:86]: That is just to get mypy to not fail type errors by giving the first reference a type.

File: doc/source/analytics/explainers.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (axsaucedo) [reply doc/source/analytics/explainers.md:14]: Ah good to know, we can link it to any other open issues if there are any, or we can just open one to track Redhat's python 3.7 ubi (I'd assume there's a couple of people already requesting it)

File: components/alibi-explain-server/alibiexplainer/explainer.py (hunk 1 above)
Comment (axsaucedo) [reply components/alibi-explain-server/alibiexplainer/explainer.py:86]: Ah fair, may be worth adding a comment so it's not removed

**Comment by seldondev** [comment]
Tue Aug 25 13:59:40 UTC 2020
The logs for [lint] [26] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2241/26.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2241 --build=26

**Comment by seldondev** [comment]
Tue Aug 25 13:59:52 UTC 2020
The logs for [pr-build] [25] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2241/25.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2241 --build=25

File: components/alibi-explain-server/alibiexplainer/proto/prediction.proto [hunk 4]
```diff
@@ -0,0 +1,130 @@
+syntax = "proto3";
```
[the remark this answers is not in the document]
Comment (cliveseldon) [reply components/alibi-explain-server/alibiexplainer/proto/prediction.proto:1]: changed to use centralized protos

**Comment by seldondev** [comment]
Tue Aug 25 14:45:32 UTC 2020
The logs for [pr-build] [27] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2241/27.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2241 --build=27

**Comment by seldondev** [comment]
Tue Aug 25 14:45:40 UTC 2020
The logs for [lint] [28] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2241/28.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2241 --build=28

File: doc/source/analytics/explainers.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (cliveseldon) [reply doc/source/analytics/explainers.md:14]: https://github.com/SeldonIO/seldon-core/issues/2324

File: components/alibi-explain-server/alibiexplainer/explainer.py (hunk 2 above)
Comment (cliveseldon) [reply components/alibi-explain-server/alibiexplainer/explainer.py:109]: As its internal calls easier to use requests directly

File: servers/sklearnserver/sklearnserver/requirements.txt [hunk 5]
```diff
@@ -1,5 +1,5 @@
-seldon_core
-scikit-learn == 0.20.3
+seldon_core[gcs,azure]
```
[the remark this answers is not in the document]
Comment (cliveseldon) [reply servers/sklearnserver/sklearnserver/requirements.txt]: removed

**Comment by seldondev** [comment]
Tue Aug 25 14:54:16 UTC 2020
The logs for [lint] [30] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2241/30.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2241 --build=30

**Comment by seldondev** [comment]
Tue Aug 25 14:54:28 UTC 2020
The logs for [pr-build] [29] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2241/29.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2241 --build=29

**Comment by seldondev** [comment]
Tue Aug 25 15:08:45 UTC 2020
The logs for [notebooks] [31] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2241/31.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2241 --build=31

File: components/alibi-explain-server/alibiexplainer/explainer.py (hunk 2 above)
Comment (axsaucedo) [reply components/alibi-explain-server/alibiexplainer/explainer.py:109]: Ok makes sense, the seldonclient is due a rewrite

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[axsaucedo](https://github.com/SeldonIO/seldon-core/pull/2241#issuecomment-680090252 "Approved")*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com#issuecomment-680090252/SeldonIO/seldon-core/blob/master/OWNERS)~~ [axsaucedo]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 9e80fa2878854f2fe9ac13a3f26a88618f0ec9e9 | [link]() | `/test integration`
notebooks | 9e935f40bfe3b7a0f8cdc666f9ad692e83414404 | [link]() | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
New changes are detected. LGTM label has been removed.

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[axsaucedo](https://github.com/SeldonIO/seldon-core/pull/2241#issuecomment-680090252 "Approved")*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [axsaucedo]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
Tue Aug 25 16:11:30 UTC 2020
The logs for [pr-build] [32] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2241/32.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2241 --build=32

**Comment by seldondev** [comment]
Tue Aug 25 16:11:32 UTC 2020
The logs for [lint] [33] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2241/33.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2241 --build=33

**Comment by seldondev** [comment]
Tue Aug 25 16:12:01 UTC 2020
The logs for [notebooks] [34] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2241/34.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2241 --build=34

**Comment by seldondev** [comment]
@cliveseldon: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 9e80fa2878854f2fe9ac13a3f26a88618f0ec9e9 | [link]() | `/test integration`
notebooks | daf62f03329fd4be4c770a5040fd68244319889e | [link]() | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
Tue Aug 25 21:01:44 UTC 2020
The logs for [pr-build] [35] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2241/35.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2241 --build=35

**Comment by seldondev** [comment]
Tue Aug 25 21:01:54 UTC 2020
The logs for [lint] [36] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2241/36.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2241 --build=36

**Comment by seldondev** [comment]
Tue Aug 25 21:02:03 UTC 2020
The logs for [notebooks] [37] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2241/37.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2241 --build=37

**Comment by seldondev** [comment]
@cliveseldon: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 9e80fa2878854f2fe9ac13a3f26a88618f0ec9e9 | [link]() | `/test integration`
notebooks | daf62f03329fd4be4c770a5040fd68244319889e | [link]() | `/test notebooks`
pr-build | 61ec2aa4a53f50bf673ad4ca7b3d6dad1eca320e | [link]() | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
Wed Aug 26 07:06:46 UTC 2020
The logs for [pr-build] [38] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2241/38.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2241 --build=38

**Comment by seldondev** [comment]
@cliveseldon: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 9e80fa2878854f2fe9ac13a3f26a88618f0ec9e9 | [link]() | `/test integration`
pr-build | 61ec2aa4a53f50bf673ad4ca7b3d6dad1eca320e | [link]() | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
Wed Aug 26 07:38:37 UTC 2020
The logs for [integration] [39] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2241/39.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2241 --build=39

**Pull request merged**